### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by sylwiabr**
Adds a basic documentation for Dataframes library.

### CODE REVIEW

File: src/Table.luna [hunk 1]
```diff
@@ -58,6 +58,20 @@ class Table:
...
+    # > import Dataframes.Column 
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    #
+    # `columns`: List of columns to construct the `Table`.
```
Comment (mikusp) [inline src/Table.luna]: Can `columns` argument be empty? What if there are two columns with the same name in the list?

File: src/Table.luna [hunk 2]
```diff
@@ -71,27 +85,128 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > column = table.columnAt 1
+    #
+    # `index`: Integer column index.
```
Comment (mikusp) [inline src/Table.luna]: Are columns indexed from 0 or from 1? What happens for out of bounds access?

File: src/Table.luna [hunk 3]
```diff
@@ -71,27 +85,128 @@ class Table:
...
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > l3 = [21,22,23,24,25]
+    # > col3 = Column.fromList "col3" Int64Type l3 
+    # > newTable = table.addColumn 1 col3
+    #
+    # `index`: Integer index for newly inserted column.
```
Comment (mikusp) [inline src/Table.luna]: Columns indexed from 0 or 1? What happens to the column that was on this position? Can you add a column on out of bounds index?

File: src/Table.luna [hunk 4]
```diff
@@ -71,27 +85,128 @@ class Table:
...
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > l3 = [21,22,23,24,25]
+    # > col3 = Column.fromList "col3" Int64Type l3 
+    # > newTable = table.setAt "foo" col3
+    #
+    # `name`: String name for newly added column.
```
Comment (mikusp) [inline src/Table.luna]: What if there is a column with that name in a table? Does column order matter? If yes, what is the algorithm for determining where to put the new column?

File: src/Table.luna [hunk 5]
```diff
@@ -71,27 +85,128 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > newTable = table.columns ["col1"] 
+    #
+    # `names`: List of column names to be returned.
```
Comment (mikusp) [inline src/Table.luna]: What if user specifies some name more than one time? What if a given name is not present in a table?

File: src/Table.luna [hunk 6]
```diff
@@ -71,27 +85,128 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > col = table.column "col1"
+    #
+    # `name`: Name of the column to return.
```
Comment (mikusp) [inline src/Table.luna]: What if a column does not exist?

File: src/Table.luna [hunk 7]
```diff
@@ -115,7 +230,39 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > take2 = table.take 2
+    #
+    # `count`: Number of rows to take.
```
Comment (mikusp) [inline src/Table.luna]: What happens on negative input or value that is bigger than number of rows?

File: src/Table.luna [hunk 8]
```diff
@@ -115,7 +230,39 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > take2 = table.drop 2
+    #
+    # `count`: Number of rows to drop.
```
Comment (mikusp) [inline src/Table.luna]: Same as with `take`

File: src/Table.luna [hunk 9]
```diff
@@ -126,21 +273,102 @@ class Table:
...
+    #
+    # > import Dataframes.Column 
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > column = table.each (_.at "col1" * 10)
```
Comment (mikusp) [inline src/Table.luna]: In `filter` similar lambda was written without underscore, it's better to keep it consistent everywhere

File: src/Table.luna [hunk 10]
```diff
@@ -126,21 +273,102 @@ class Table:
...
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > newTable = table.eachTo "col3" (_.at "col1" * 10)
+    #
+    # `name`: Name of newly created column in result table.
+    # `f`: Function to apply on `self` table.
+    # `return`: `Table` value with `self` table and additionla column with provided name and values resulting from function `f` application.
```
Comment (mikusp) [inline src/Table.luna]: additional

Comment (mikusp) [reply src/Table.luna]: `self` with additional column `name` containing values returned by `f` for each row

File: src/Table.luna [hunk 11]
```diff
@@ -152,21 +380,100 @@ class Table:
...
         sortByColumns = sortBy.each (colName, order, pos): (self.column colName, order, pos)
         self.sortByColumn sortByColumns
+
+    # Remove rows with missing values.
+    #
+    # > import Dataframes.Column 
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,,13,14,15]
```
Comment (mikusp) [inline src/Table.luna]: Is this proper Luna syntax?

File: src/Table.luna [hunk 12]
```diff
@@ -152,21 +380,100 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,,5]
+    # > l2 = [11,,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > newTable = table.describeNa
+    #
+    # `return`: `Table` value with statistics of NA/NaN values for each column.
```
Comment (mikusp) [inline src/Table.luna]: What are the names of columns in `return`?

File: src/Table.luna [hunk 13]
```diff
@@ -181,10 +488,38 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,,5]
+    # > l2 = [11,,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > correlations = table.corr
+    #
+    # `return`: `Table` value with pairwise correlations calculated for `self` table.
```
Comment (mikusp) [inline src/Table.luna]: What are the names of columns in `return`?

File: src/Table.luna [hunk 14]
```diff
@@ -181,10 +488,38 @@ class Table:
...
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,,5]
+    # > l2 = [11,,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > correlations = table.corrWith "col2"
+    #
+    # `columnName`: Column for which the correlations are calculated.
+    # `return`: `Table` value including correlations with `columnName` column calculated for `self` table.
```
Comment (mikusp) [inline src/Table.luna]: What are the names of columns in `return`?

File: src/Table.luna [hunk 15]
```diff
@@ -198,7 +533,23 @@ class Table:
         col2 = Column.fromList ("CORR_WITH_" + columnName) DoubleType vals
         Table.fromColumns [col1, col2]
 
-    # describeColumn :: Text -> Table 
+    # Generates descriptive statistics that summarize the central tendency, dispersion and shape of the values in given column distribution. 
+    #
+    # Analyzes just numeric values, excluding NA/null values. Returns statistics like: mean, standard deviation, minimum, 1st quantile, median, 3rd quantile, maximum.
```
Comment (mikusp) [inline src/Table.luna]: Like these or exactly these?

File: src/Table.luna [hunk 16]
```diff
@@ -198,7 +533,23 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > correlations = table.describeColumn "col2"
+    #
+    # `columnName`: String value, name of the column to calculate statistics for.
```
Comment (mikusp) [inline src/Table.luna]: What happens if a column with type other than numeric is specified?

File: src/Table.luna [hunk 17]
```diff
@@ -267,33 +631,113 @@ class Table:
...
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > plot1 = table.chart.plot "col1" "col2"
+    #
+    # `columnName1`: X column name.
+    # `columnName2`: Y column name.
+    # `return`: [] OBRAZEK
```
Comment (mikusp) [inline src/Table.luna]: English even in placeholders

File: src/Table.luna [hunk 18]
```diff
@@ -267,33 +631,113 @@ class Table:
 class ChartBuilder:
     table :: Table
 
-    def plot cname1 cname2:
-        c1 = self.table.at cname1 . toColumn
-        c2 = self.table.at cname2 . toColumn
+    # Plot y versus x as markers or lines.
```
Comment (mikusp) [inline src/Table.luna]: or?

File: src/Table.luna [hunk 19]
```diff
@@ -267,33 +631,113 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > plot1 = table.chart.kde2 "col1" "col2"
+    #
+    # `columnName1`: First input data.
```
Comment (mikusp) [inline src/Table.luna]: Not data but column name. What happens if column specified is not numeric or does not exist?

File: src/IO.luna [hunk 20]
```diff
@@ -71,6 +71,14 @@ class CSVParser:
         CustomNames l: l
         _: []
 
+    # Read CSV file from the hard drive, returning a `Table` value.
+    # 
+    # > import Dataframes.IO 
+    # > CSVParser.readFile "./data/simple_empty.csv"
+    #
+    # `filepath`: Path to the CSV file to read.
```
[the remark this answers is not in the document]
Comment (sylwiabr) [reply src/IO.luna]: It should be done automatically not manually.

File: src/Table.luna (hunk 18 above)
Comment (sylwiabr) [reply src/Table.luna]: you can set styles for `plot` plot

File: src/Table.luna (hunk 4 above)
Comment (sylwiabr) [reply src/Table.luna]: new column is always at the end of the table

File: src/IO.luna [hunk 21]
```diff
@@ -71,6 +71,14 @@ class CSVParser:
...
+    # `return`: Table value containing the data from CSV file.
```
[the remark this answers is not in the document]
Comment (piotrMocz) [reply src/IO.luna]: Yeah, that's why the python docs explicitly use the `param` and `returns` keywords.

File: src/Table.luna (hunk 4 above)
Comment (piotrMocz) [reply src/Table.luna]: That's worth adding to the documentation ;)

File: src/Table.luna (hunk 6 above)
Comment (piotrMocz) [reply src/Table.luna]: This is a confusing choice of words, I'm afraid. Is the table located under this given name? That's how I would understand it. Maybe we could go with something like `Lookup a Column by name`? The `Table` part can be figured out from the context quite easily. This applies to the whole family of indexing/adding/removing methods.

File: src/Table.luna (hunk 12 above)
Comment (piotrMocz) [reply src/Table.luna]: "ratio" -> "percentage" :)

File: src/Table.luna (hunk 15 above)
Comment (piotrMocz) [reply src/Table.luna]: "Analyzes just **the** numeric values".

File: src/Table.luna (hunk 16 above)
Comment (piotrMocz) [reply src/Table.luna]: the statistics

File: src/Table.luna (hunk 13 above)
Comment (piotrMocz) [reply src/Table.luna]: the self table

File: src/Table.luna (hunk 10 above)
Comment (piotrMocz) [reply src/Table.luna]: Also: "as a result of applying the function `f` to each row of the `self` table"

File: src/Table.luna [hunk 22]
```diff
@@ -126,21 +283,107 @@ class Table:
...
+    # `pred`: {min, max, mean, median, std, var, sum}.
+    #
+    # `return`: Table with function application results.
+
     def eachColumn pred:
         cols = self.toList
         filtered = cols.each pred
         self.fromColumns filtered
+
+    # Returns `Column` value with results of function `f` application.
```
Comment (piotrMocz) [inline src/Table.luna]: Returns a `Column` created as a result of applying the function `f` to each row of the table.

File: src/Table.luna [hunk 23]
```diff
@@ -152,21 +395,78 @@ class Table:
     def sort sortBy:
         sortByColumns = sortBy.each (colName, order, pos): (self.column colName, order, pos)
         self.sortByColumn sortByColumns
+
+    # Remove rows with missing values.
```
Comment (piotrMocz) [inline src/Table.luna]: Please keep the mood consistent. I prefer the imperative but we keep using indicative throught these docs.

This applies to all of the headers.

File: src/Table.luna [hunk 24]
```diff
@@ -152,21 +395,78 @@ class Table:
...
+    # > import Dataframes.IO 
+    # > table = CSVParser.readFile "./data/simple_empty.csv"
+    # > newTable = table.dropNa
+    #
+    # `return`: `Table` value without rows containing NA records.
+
     def dropNa:
         self.fromWrapper $ self.ptr.dropNa
+
+    # Remove rows where missing values occurs for given column.
```
Comment (piotrMocz) [inline src/Table.luna]: occurs -> occur
for **a** given column

File: data/simple_empty.csv [hunk 25]
```diff
@@ -1,5 +1,5 @@
 a,b,c
-1,2,3
+1,2,
```
Comment (mwu-tow) [inline data/simple_empty.csv]: Is this change intended?

File: src/IO.luna [hunk 26]
```diff
@@ -71,6 +71,15 @@ class CSVParser:
         CustomNames l: l
         _: []
 
+    # Read CSV file from the hard drive, returning a `Table` value.
+    # 
+    # > import Dataframes.IO 
+    # > CSVParser.readFile "./data/simple_empty.csv"
```
Comment (mwu-tow) [inline src/IO.luna]: There should be information that by default:
1) first row is assumed to contain column names
2) column types are deduced

File: src/Main.luna [hunk 27]
```diff
@@ -1,12 +1,13 @@
-import Std.Base
-import Std.Foreign.C.Value
-
-import Dataframes.Internal.Utils
-import Dataframes.Array
+import Dataframes.Table
 import Dataframes.Types
-import Dataframes.Internal.Test.Test
-
+import Dataframes.Column
```
Comment (mwu-tow) [inline src/Main.luna]: Is this intended part of PR?

File: src/Table.luna (hunk 1 above)
Comment (mwu-tow) [reply src/Table.luna]: The only tricky edge case that can break things is passing columns with different rows count. We should decide what to do (throw? trim to shortest? pad with nulls?) — and implement and document this.

File: src/Table.luna [hunk 28]
```diff
@@ -71,27 +86,135 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > column = table.at "col1"
+    #
+    # `name`: Label for column.
```
Comment (mwu-tow) [inline src/Table.luna]: we should unify terminology: label vs name

File: src/Table.luna [hunk 29]
```diff
@@ -71,27 +86,135 @@ class Table:
...
+    # `return`: `Table` value containing previous `table` with newly inserted column.
+
     def addColumn index column:
         self.fromWrapper $ self.ptr.addColumn index column.ptr
     def concat ts:
         cols = ts . each .toList . concat
         self.fromColumns cols
     def + that: Table.concat [self, that]
+
+    # Returns a `Table` value contain column matching given name.
```
Comment (mwu-tow) [inline src/Table.luna]:
```
Returns a `Table` value containing a single column matching the given name.
```

File: src/Table.luna [hunk 30]
```diff
@@ -71,27 +86,135 @@ class Table:
...
+    #
+    # `name`: Name of the column to return.
+    #
+    # `return`: Single `Column` value selected from `self` table by name.
+
     def column name:
         ix = self.schema.getFieldIndexByName name
         self.columnAt ix    
+    
+    # Returns `Table` value being a subset of `self` table containing given list of columns.
```
Comment (mwu-tow) [inline src/Table.luna]: "subset of Table"  sounds strange to me.
Returns a `Table` value with subset of columns from `self` designated by the given name list.

File: src/Table.luna (hunk 5 above)
Comment (mwu-tow) [reply src/Table.luna]: And what if there's is more than a single column matching the given name?

File: src/Table.luna (hunk 4 above)
Comment (mwu-tow) [reply src/Table.luna]: Why "String name"? There are no other kinds of names. I'd say just "Name".
Or, if we want to include type information, it should be clear what is the typename (Text).

File: src/Table.luna [hunk 31]
```diff
@@ -126,21 +283,107 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" Int64Type l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > filteredTable = table.filter (x: x.at "col2" < 14)
+    #
+    # `f`: The predicate (Boolean-valued function).
```
Comment (mwu-tow) [inline src/Table.luna]: I see two problems with "(Boolean-valued function).":
1) it looks like typename ersatz — we should either omit it or put the proper Luna type
2) strictly speaking this expression does not yield Bool anywhere

The most surprising thing (and needing explanation) for user is what is the type of "x" and what operations are allowed.

File: src/Table.luna [hunk 32]
```diff
@@ -126,21 +283,107 @@ class Table:
...
+    # > import Dataframes.Types
+    # > import Dataframes.Table 
+    # > l1 = [1,2,3,4,5]
+    # > l2 = ["a", "b", "c", "d", "e"]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > filteredTable = table.filterColumns .isNumeric
+    #
+    # `f`: The predicate (Boolean-valued function).
```
Comment (mwu-tow) [inline src/Table.luna]: as above

File: src/Table.luna [hunk 33]
```diff
@@ -126,21 +283,107 @@ class Table:
...
+    #
+    # `return`: Table with columns satisfying predicate.
+
     def filterColumns pred:
         cols = self.toList
         filtered = cols.filter pred
         self.fromColumns filtered
+    
+    # Returns `Table` value with results of applied functions on each of the columns from `self` table.
```
Comment (mwu-tow) [inline src/Table.luna]: Preconditions upon the columns should be stated. Type must be numeric, taking a mean of String column is an error.

File: src/Table.luna [hunk 34]
```diff
@@ -126,21 +283,107 @@ class Table:
...
+    # > l1 = [1,2,3,4,5]
+    # > l2 = [11,12,13,14,15]
+    # > col1 = Column.fromList "col1" Int64Type l1
+    # > col2 = Column.fromList "col2" StringType l2
+    # > table = Table.fromColumns [col1 , col2]
+    # > filteredTable = table.eachColumn .max
+    #
+    # `pred`: {min, max, mean, median, std, var, sum}.
+    #
+    # `return`: Table with function application results.
```
Comment (mwu-tow) [inline src/Table.luna]: It should be explained how the "results" look like.

File: src/Table.luna [hunk 35]
```diff
@@ -152,21 +395,78 @@ class Table:
...
         self.sortByColumn sortByColumns
+
+    # Remove rows with missing values.
+    #
+    # > import Dataframes.Table
+    # > import Dataframes.IO 
+    # > table = CSVParser.readFile "./data/simple_empty.csv"
+    # > newTable = table.dropNa
+    #
+    # `return`: `Table` value without rows containing NA records.
```
Comment (mwu-tow) [inline src/Table.luna]: We need to unify terms:
* NA
* missing value
* null
* NAN
* hole

The Arrow's term is "null value". Certainly better than "nan" that makes sense only for numeric columns.

File: src/Table.luna [hunk 36]
```diff
@@ -152,21 +395,78 @@ class Table:
...
+    # Remove rows where missing values occurs for given column.
+    #
+    # > import Dataframes.Table
+    # > import Dataframes.IO 
+    # > table = CSVParser.readFile "./data/simple_empty.csv"
+    # > newTable = table.dropNaAt "b"
+    #
+    # `name`: Column name for which we check NA values.
+    #
+    # `return`: `Table` value without rows containing NA records in given column.
```
Comment (mwu-tow) [inline src/Table.luna]: not sure about "record" usage here

File: src/Table.luna [hunk 37]
```diff
@@ -219,7 +565,20 @@ class Table:
         aggregationWithWrappers = aggregations.each (colname, aggrs): (self.column colname . ptr, aggrs.each .toInt)
         self.fromWrapper $ self.ptr.aggregateBy keyColumnWrapper aggregationWithWrappers
 
-    # describe returns informations about whole table like standard deviation, minium, maximum, mean, median qunatile 0.25 and 0.75 for each column
+    # Describe returns informations about whole table like standard deviation, minium, maximum, mean, median qunatile 0.25 and 0.75 for each column.
```
Comment (mwu-tow) [inline src/Table.luna]: typo: quantile

Comment (mwu-tow) [reply src/Table.luna]: also: comma after "median"

File: src/Table.luna [hunk 38]
```diff
@@ -126,21 +283,107 @@ class Table:
...
+    # `f`: Function to apply on `self` table.
+    #
+    # `return`: `Column` value with function application results.
+
     def each f:
         mapper = f TableHandle
         columnWrapper = self.ptr.eachToColumn "Result" mapper
         Column.fromColumnWrapper columnWrapper
+    
+    # Adds to `self` table a column with `name` and values as a result of application function `f` on the `self` table.
```
Comment (mwu-tow) [inline src/Table.luna]: "adds to self" sounds to me as if self got mutated
Creates a table with columns from `self` and an additional column named `name` being a result of… etc

File: src/Table.luna (hunk 10 above)
Comment (mwu-tow) [reply src/Table.luna]: Do we need to duplicate that much information? Return description is almost the same as function synopsis.

File: src/Table.luna (hunk 6 above)
Comment (sylwiabr) [reply src/Table.luna]: @piotrMocz we have very similar methods but some of them returns columns and some tables: so we have `Table.column "foo"` and it will return column `foo` and we have method `Table.at "foo"` which will give you Table with single column `foo` this is why I describe over and over returns values in a short description

File: data/simple_empty.csv (hunk 25 above)
Comment (sylwiabr) [reply data/simple_empty.csv]: yes

File: src/Table.luna (hunk 29 above)
Comment (sylwiabr) [reply src/Table.luna]: @mwu I have fixed the code. About the second part: since we want to have full example with fully working code in documentation for user to copy it from searcher I don't see how we can avoid code repetition? We can make bigger example to copy each time but then we need somehow say to parser and searcher that it is at some place. The actual values result should be generated from code sample but since we don't have such tools I am adding screens with the outputs.

File: src/Table.luna (hunk 36 above)
Comment (sylwiabr) [reply src/Table.luna]: So how would we name single cell value in table? Any ideas? @piotrMocz @mikusp

File: src/Table.luna (hunk 29 above)
Comment (mwu-tow) [reply src/Table.luna]: @sylwiabr I don't have good answer here. I just feel that either our requirements are wrong or we sorely miss necessary tooling. Doing all this by hand is very time-consuming and will impose serious maintanance burden in future. Particularly when we plan to significantly revamp the API. Perhaps we should discuss this in more detail next monday. This issue is likely out of scope for this PR, we rather need to have some bigger-scale discussion on how to best approach documentation generation.

Comment (sylwiabr) [reply src/Table.luna]: With this I agree. Just don't have right solution at this moment and we need desperately any docs with examples.

File: src/Table.luna [hunk 39]
```diff
@@ -121,7 +298,55 @@ class Table:
         cols = self.toList
         slicedCols = cols.each (_.slice beg count)
         self.fromColumns slicedCols
+    
+    # Returns the first `count` rows
+    #
+    # If the count of elements to take is greater than actual rows count, the returned table is tha same as the `self` table. 
+    # For `count` number negative the tabel with no data will be returned.
```
Comment (mikusp) [inline src/Table.luna]: `the table`

File: src/Table.luna [hunk 40]
```diff
@@ -132,21 +357,130 @@ class Table:
...
+    # > def main:
+    # >     l1 = [1,2,3,4,5]
+    # >     l2 = [11,12,13,14,15]
+    # >     col1 = Column.fromList "col1" Int64Type l1
+    # >     col2 = Column.fromList "col2" Int64Type l2
+    # >     table = Table.fromColumns [col1 , col2]
+    # >     filteredTable = table.filter (x: x.at "col2" < 14)
+    # >     None
+    #
+    # `f`: The LQuery predicate. [http://docs.luna-lang.org/#Std.Base] --- fixme proper link
```
Comment (mikusp) [inline src/Table.luna]: Proper link? Or none at all.

File: src/Table.luna (hunk 13 above)
Comment (mikusp) [reply src/Table.luna]: Not resolved here

File: src/Table.luna [hunk 41]
```diff
@@ -227,7 +730,24 @@ class Table:
         aggregationWithWrappers = aggregations.each (colname, aggrs): (self.column colname . ptr, aggrs.each .toInt)
         self.fromWrapper $ self.ptr.aggregateBy keyColumnWrapper aggregationWithWrappers
 
-    # describe returns informations about whole table like standard deviation, minium, maximum, mean, median qunatile 0.25 and 0.75 for each column
+    # Describes returns informations about whole table: standard deviation, minium, maximum, mean, median qunatile 0.25 and 0.75 for each column.
```
Comment (mikusp) [inline src/Table.luna]: `returns`, `minium`
